nxcomp/Log.h

```diff
@@ -198,18 +286,68 @@ NXLog& operator<< (NXLog& out, const NXLogStamp& value);
 
 
 template <typename T>
+bool has_newline(T value)
+{
+	return false;
+}
+
+template <char*>
+static bool has_newline(char *value)
+{
+	return strstr(value, "\n") != NULL;
```

Comment:
Just to clarify: I don't mean to improve performance by using the other explicit implementation here, but drop one specific implementation and thus have one specific implementation less to maintain in the long run. It won't make a really huge difference, as these functions are very short and easy to understand as-is, but enforcing code reuse if possible is a good thing.